### PR TITLE
Change post-London fee logic so that all trx types burn fees

### PR DESCRIPTION
**Pull request by YordanPavlov**
A PR to fix the fee burning logic.
1. Even old style transactions burn the base fee after the London fork.
2. For the new type 2 transactions check if maxFeePerGas is not reducing the fee paid.

### CODE REVIEW

File: blockchains/eth/lib/constants.js [hunk 1]
```diff
@@ -3,11 +3,13 @@ const CONFIRMATIONS = parseInt(process.env.CONFIRMATIONS || "3")
 const PARITY_NODE = process.env.PARITY_URL || "http://localhost:8545/"
 const LOOP_INTERVAL_CURRENT_MODE_SEC = parseInt(process.env.LOOP_INTERVAL_CURRENT_MODE_SEC || "30")
 const BURN_ADDRESS = "burn"
+const LONDON_FORK_BLOCK = 12965001
```
Comment (nailkhaf) [inline blockchains/eth/lib/constants.js:6]: why not 12965000?
https://etherscan.io/block/12965000

File: blockchains/eth/lib/fees_decoder.js [hunk 2]
```diff
@@ -12,52 +11,114 @@ class FeesDecoder {
...
+     *
+     * EIP1559 transactions must specify both maxPriorityFeePerGas and maxFeePerGas. They must not specify gasPrice.
+     *
+     * https://besu.hyperledger.org/en/stable/Concepts/Transactions/Transaction-Types/
+     */
+    const maxPriorityFeePerGas = this.web3Wrapper.parseValueToBN(transaction['maxPriorityFeePerGas'])
+    const maxFeePerGas = this.web3Wrapper.parseValueToBN(transaction['maxFeePerGas'])
+    const baseFeePerGas = this.web3Wrapper.parseValueToBN(block.baseFeePerGas)
+
+    const minerFeePerGas = baseFeePerGas.add(maxPriorityFeePerGas).gt(maxFeePerGas) ?
```
Comment (nailkhaf) [inline blockchains/eth/lib/fees_decoder.js:50]: `transaction['gasPrice'] - block.baseFeePerGas`

When I query mined tx type 2 I got field gasPrice, isn't this the same?

Comment (YordanPavlov) [reply blockchains/eth/lib/fees_decoder.js]: Thanks for this observation. I did not read the whole EIP and thought that the 'gasPrice' field is a leftover to be removed. You are right that it must be present and we can depend on it.

**Review by YordanPavlov — COMMENTED** [review]
Thanks for your input. I've addressed the recommendations.

**Review by nailkhaf — APPROVED** [review]

**Pull request merged**